frame/sudo/src/mock.rs

```diff
@@ -20,52 +20,95 @@ use super::*;
 
 use frame_support::{
 	impl_outer_origin, impl_outer_dispatch, impl_outer_event, parameter_types,
-	weights::Weight,
-	// traits::{OnInitialize, OnFinalize},
+	weights::{Weight, DispatchClass}
 };
 use sp_core::H256;
 // The testing primitives are very useful for avoiding having to work with signatures
 // or public keys. `u64` is used as the `AccountId` and no `Signature`s are required.
-use sp_runtime::{
-	Perbill, traits::{BlakeTwo256, IdentityLookup}, testing::Header,
-};
+use sp_runtime::{Perbill, traits::{BlakeTwo256, IdentityLookup}, testing::Header};
 use sp_io;
-
 use crate as sudo;
 
-impl_outer_origin! {
-	pub enum Origin for Test where system = frame_system {}
-}
-impl_outer_event! {
-	pub enum TestEvent for Test {
-		system<T>,
-		sudo<T>,
+// logger module used by privileged_function() to track execution
+mod logger {
+	use super::*;
+	use std::cell::RefCell;
+	use frame_system::ensure_root;
+
+	thread_local! {
+		static LOG: RefCell<Vec<u32>> = RefCell::new(Vec::new());
 	}
-}
-impl_outer_dispatch! {
-	pub enum Call for Test where origin: Origin {
-		sudo::Sudo,
-		priveleged_fn_test_module::Priveleged,
+	pub fn log() -> Vec<u32> {
+		LOG.with(|log| log.borrow().clone())
+	}
+	pub trait Trait: system::Trait {
+		type Event: From<Event> + Into<<Self as system::Trait>::Event>;
+	}
+	decl_storage! {
+		trait Store for Module<T: Trait> as Logger {
+		}
+	}
+	decl_event! {
+		pub enum Event {
+			Logged(u32, Weight),
+		}
+	}
+	decl_module! {
+		pub struct Module<T: Trait> for enum Call where origin: <T as system::Trait>::Origin {
+			fn deposit_event() = default;
+
+			#[weight = FunctionOf(
+				|args: (&u32, &Weight)| *args.1,
+				|_: (&u32, &Weight)| DispatchClass::Normal,
+				Pays::Yes,
+			)]
+			fn log(origin, i: u32, weight: Weight) {
+				ensure_root(origin)?;
+				Self::deposit_event(Event::Logged(i, weight));
+				LOG.with(|log| {
+					log.borrow_mut().push(i);
+				})
+			}
+		}
 	}
 }
 
 // Dummy module with a privelleged dispatchable function for testing sudo
 mod priveleged_fn_test_module {
 	use frame_support::{decl_module, dispatch};
 	use frame_system::ensure_root;
+
 	pub trait Trait: frame_system::Trait {}
 
 	decl_module! {
 		pub struct Module<T: Trait> for enum Call where origin: T::Origin {
 			#[weight = 0]
 			pub fn privileged_function(origin) -> dispatch::DispatchResult {
 				ensure_root(origin)?;
-				println!("This is priveleged_fn_test_module::privileged_function() doing something.");
+				// Log the origin and an arbitrary weight of 42
+				// logger::call::log(origin, 42);
```

Comment:
This doesnt make sense. You should only keep the logger, and remove this dummy module. You only need one of the two

Reply:
The dummy module has been removed. All tested sudo functions that require a call are now passed a call from the logger module.